### PR TITLE
feat: add sheriff dry run

### LINKED ISSUE #29: Run a dry-run mode permissions job on PRs to the .permissions repo
We should run a dry-run job on every `.permissions` PR and put the logs / changes in the check job.  If the dry-run fails we should make the check a hard fail.

This will replace the somewhat hacked together https://github.com/codebytere/governance-check module.  We should ensure all the safety checks / helpful errors in that module are converted / available in the new check run.

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Closes #29

This runs a PR dry run check against the `.permissions` repo on every PR

### CODE REVIEW

File: src/dry-run-q.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      const gistOctokit = new Octokit({
+        auth: SHERIFF_GIST_TOKEN,
+      });
+      const gistData = await gistOctokit.gists.create({
+        files: {
+          ['out.svg']: {
+            content: svg,
+          },
+        },
+      });
```
Comment (dsanders11) [inline src/dry-run-q.ts:111]: Where does this Gist live when a GH app is running this code? The API documentation only lists "GitHub App user access tokens", not installation tokens.

File: src/dry-run-q.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        owner: PERMISSIONS_FILE_ORG,
+        name: SHERIFF_DRY_RUN_CHECK_NAME,
+        head_sha: headSha,
+        status: 'completed',
+        conclusion: success ? 'success' : 'failure',
+        completed_at: new Date().toISOString(),
+        output: {
+          title: 'Dry Run Output',
+          summary: success ? 'Looking good' : "Something isn't looking so hot",
+          text: `<img src="${svgUrl}" width="800" />`,
```
Comment (dsanders11) [inline src/dry-run-q.ts:125]: Is the SVG for aesthetic reasons, or to work around the max length for the `text` field?

**Review by dsanders11 — CHANGES_REQUESTED** [review]
Flagging a concern about if Gist creation will work without an associated user.

File: src/dry-run-q.ts (hunk 1 above)
Comment (dsanders11) [reply src/dry-run-q.ts:111]: I guess that's what `SHERIFF_GIST_TOKEN` is for? Can that be documented?

File: src/dry-run-q.ts (hunk 2 above)
Comment (MarshallOfSound) [reply src/dry-run-q.ts:125]: Both, it looks better and the text limit is arbitrary and annoying 🙃

File: src/dry-run-q.ts (hunk 1 above)
Comment (MarshallOfSound) [reply src/dry-run-q.ts:111]: > I guess that's what SHERIFF_GIST_TOKEN is for? 

yeah, I don't love it, but wanted to get this working before wasting too many cycles figuring out a better way to store SVG files 🙃

**Review by dsanders11 — APPROVED** [review]

File: src/dry-run-q.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        repo: PERMISSIONS_FILE_REPO,
+        owner: PERMISSIONS_FILE_ORG,
+        name: SHERIFF_DRY_RUN_CHECK_NAME,
+        head_sha: headSha,
+        status: 'completed',
+        conclusion: success ? 'success' : 'failure',
+        completed_at: new Date().toISOString(),
+        output: {
+          title: 'Dry Run Output',
+          summary: success ? 'Looking good' : "Something isn't looking so hot",
```
[the remark this answers is not in the document]
Comment (dsanders11) [reply src/dry-run-q.ts]: Whoops 😅 That's what I get for making code suggestions late at night.

**Review by VerteDinde — APPROVED** [review]
Nice!

**Pull request merged**